src/duplo_resource/asg.py

```diff
@@ -14,8 +14,12 @@ def __init__(self, duplo: DuploClient):
   def list(self):
     """Retrieve a list of all services in a tenant."""
     tenant_id = self.tenant["TenantId"]
+    tenant_name = self.tenant["AccountName"]
     response = self.duplo.get(f"subscriptions/{tenant_id}/GetTenantAsgProfiles")
-    return response.json()
+    if (data := response.json()):
```

Comment:
The _validate_response function in client.py https://github.com/duplocloud/duploctl/blob/b5cb39ab0668b0736a1616081c2fb5e3971b1b8e/src/duplocloud/client.py#L199 is already checking for 404s.  Can we not rely on the response code for cases where the resource is not found?

Validate sometimes returns `{"message": response.text}` instead of the response object, which I believe would cause issues in this check. The functions you updated should return valid JSON (I think), but is this error checking pattern reliant on the developer understanding whether validate should return the response or the `{"message": response.text}`?

Reply:
I agree with Matt here. The validate response is already handling the 404 cases and therefore the else statement here will never run.